rfcs/0002-typography-scaling.md

```diff
@@ -0,0 +1,162 @@
+- **Does not cater for landing/marketing pages.** Which are generally much larger compared to the size-ratio on the dashboard (Maybe a [Major Third](https://type-scale.com/?size=14&scale=1.125&text=A%20Visual%20Type%20Scale&font=Lato&fontweight=400&bodyfont=Poppins&bodyfontweight=400&lineheight=1.65&backgroundcolor=white&fontcolor=%23333&preview=false) scale will work better). This is currently being looked by the comm. design team based on the same principles.
+- **Same type scale across all devices.** After the type scale is defined, we also need to define the scale across multiple device sizes. There are few approaches that we've came up with and after research from other DS/products like *GitHub*, *Eightshapes*, *Paystack*, *Stripe*, and a few more we've observed that for dashboards generally tweaking the **heading** sizes works best and the base font-size will remain same for all other devices/breakpoints.
+
+We'll look into the responsive type scale and see what approach fits best for our use case.
+
+### **Our Approach**
+>**Similar vertical & horizontal ratio across multiple breakpoints (for headings)**
+
+Our product is majorly used on mobiles/tablets/desktops. Therefore, We'll  consider only two breakpoints for the typography scale, one for mobiles & one for both desktops & tablets. Which is for devices having width `>=991px` (approx.)
+
+![Scaling: Similar vertical & horizontal ratio](https://miro.medium.com/max/2000/1*LpBKPTjjtW3I3xKEcae--Q.png)
+
+In this approach, we use the same ratio of `1.125` down each page and across each breakpoint. Pretty straightforward, we multiply everything down and across by the same ratio.
+
+### **Exception**
+This only applies to heading styles as we don't want to change our base font size (`14px`) which remains consistent across all devices/breakpoints.
+Simply put, heading styles for mobile devices will be always in decrements of `1.125` times that of the current desktop/tablet heading style.
+
+### **Example:**
+
+| Style  | Mobile | Desktops/Tablets |
+|--------|--------|------------------|
+| `<h6>` | 14px   | 16px             |
+| `<h5>` | 16px   | 18px             |
+| `<h4>` | 18px   | 20px             |
+| `<h3>` | 20px   | 25px             |
+| `<h2>` | 25px   | 32px             |
+| `<h1>` | 32px   | 40px             |
+
+That means, for devices like, **desktops/tablets** if `<h6> = 16px` than, for **mobiles** `<h6> = (16 / 1.125) = 14px`.
+
+>📝 Note: Here we're rounding-off all the numbers for the sake of simplicity and to align the content according to the grid-system.
+
+# Drawbacks/Constraints
+Few things to consider here:
+- As we are moving towards adaptive-layout so we need to consider - should we also support responsive layout? For both cases, I don't see any challenges as the typography system remains the same.
```

Comment:
By responsive layout here do you mean that typography would also scale responsively? For example `h1` varying between 32px and 40px progressively based on screen width?

Reply:
Typography will scale responsively, given that it will only scale in segments. Which means if the `h1` at `1440px` is `40px` than at `991px` it will become `32px` directly rather than going from `**40px** -> 38px -> 36px -> 34px -> **32px**`. I hope you get my point @divyanshu013.